main.go

```diff
@@ -60,6 +60,10 @@ type Config struct {
 	CacheTimeCatchAll   time.Duration `config:"cache_time.catch_all"`
 }
 
+type Indexer interface {
+	GetPackages(context.Context, *util.GetPackagesOptions) (util.Packages, error)
+}
+
```

Comment:
This is just a loose suggestion, I don't want to interfere or stop you from making the progress.

I guess you can extract all interfaces and domain classes to a separate package, so they can be used to implement more sophisticated Indexers and Storages (more or less similarly to https://github.com/elastic/beats/tree/master/libbeat/beat).

Reply:
I like to have the interfaces defined where they are used :slightly_smiling_face: Implementations don't really need it to implement it.

If we find that there may be more users of Indexers, we can reconsider and move the interface to a separate package (an alternative can also be to duplicate the definition, what can be fine in many cases too).